File: Downloads/appecologica-main/backend/src/main/java/com/appecologica/backend/model/Vehiculo.java
package com.appecologica.backend.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "vehiculos")
public class Vehiculo {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true)
    @NotBlank
    private String patente;

    private String modelo;

    @NotNull
    private Integer capacidadKg;

    private boolean habilitado = true;
}

File: Downloads/appecologica-main/appecologica-main/app/src/main/java/com/example/gestionresiduos/model/Vehiculo.kt
package com.example.gestionresiduos.model

// Vehículo asignado a un chofer o ruta
data class Vehiculo(
    val id: String,
    val patente: String,
    val modelo: String,
    val capacidadKg: Int,
    val habilitado: Boolean = true
)
